kustomize/commands/edit/set/setsecret.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2023 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package set
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"golang.org/x/exp/slices"
+	"sigs.k8s.io/kustomize/api/ifc"
+	"sigs.k8s.io/kustomize/api/konfig"
+	"sigs.k8s.io/kustomize/api/resource"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kustomize/v5/commands/internal/kustfile"
+	"sigs.k8s.io/kustomize/kustomize/v5/commands/internal/util"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+func newCmdSetSecret(
+	fSys filesys.FileSystem,
+	ldr ifc.KvLoader,
+	rf *resource.Factory,
+) *cobra.Command {
+	var flags util.ConfigMapSecretFlagsAndArgs
+	cmd := &cobra.Command{
+		Use:   "secret NAME [--from-literal=key1=value1] [--namespace=namespace-name] [--new-namespace=new-namespace-name]",
+		Short: fmt.Sprintf("Edits the value for an existing key for a secret in the %s file", konfig.DefaultKustomizationFileName()),
+		Long: fmt.Sprintf(`Edits the value for an existing key in an existing secret in the %s file.
+Both secret name and key name must exist for this command to succeed.`, konfig.DefaultKustomizationFileName()),
```

Comment:
the length of short and long seems like same, can we remove one of them?

Reply:
Long continues on the next line! Do you feel we could add more to it?
I'd prefer to have a more complete description on the long one than scrap it completely.

Reply:
would prefer more complete in the view of a user

Reply:
Do you have an example of the format you'd like to see?